src/main/java/ch/njol/skript/expressions/ExprIndicesOf.java

```diff
@@ -0,0 +1,214 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.lang.Variable;
+import ch.njol.skript.util.LiteralUtils;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * @author Peter Güttinger
+ */
+@Name("Index Of")
+@Description({
+	"The first, last or all indices of a character (or text) in a text, or -1 if it doesn't occur in the text."
+		+ "Indices range from 1 to the <a href='#ExprIndicesOf'>length</a> of the text.",
+	"You can also get the indices or positions of a list where the value at that index is the provided value. "
+		+ "Indices are only supported for variable lists and will return the string indices of the given value. "
```

Comment:
maybe reword this to make it clear that this doesn't apply to string on string violence